src/main/java/seedu/address/logic/commands/cardcommands/AddCommand.java

```diff
@@ -169,9 +167,13 @@ public boolean equals(Object other) {
 
             // state check
             AddCommand.AddCardDescriptor e = (AddCommand.AddCardDescriptor) other;
+            if (getTag() != null) {
```

Comment:
you're right. but its just because for `null` we can't use `.equals` as will invoke a `NullPointerException`. https://stackoverflow.com/questions/4501061/java-null-check-why-use-instead-of-equals. Perhaps I can add `.getTag() === .getTag()` under the else condition for clarify, or even make it into a ternary for neatness. 

something like that

```
return getQuestion().equals(e.getQuestion())
                        && getAnswer().equals(e.getAnswer())
                        && getTag() == null ? getTag() == e.getTag() : getTag().equals(e.getTag());
```

Reply:
ahh I didnt catch that, thanks for clarifying